### DISCUSSION TITLE
Header files (.h) don't follow config

### DISCUSSION THREAD

**Discussion by gruelingpine185** [helix-editor/helix#9555] [Troubleshooting]
Say I have the following header file for a C program:

```c
#ifndef EXO_RENDERER_H
#define EXO_RENDERER_H

struct example_struct {
  const char* app_name;
  const char* engine_name;
  // ^-------  I expect everything within a scope of some sort (eg. `{ ... }`) to
  //           indent according to the config file. This means indentation of 4 space tabs
  //           rather than 2.
};

// ...

#endif // EXO_RENDERER_H
```

However, within a C file (`.c`) nothing out of the ordinary occurs regarding my configuration below. It indents by 4 spaces, which is what I want it to do.

My config file:

<img width="414" alt="Screenshot 2024-02-06 at 1 16 55 PM" src="https://github.com/helix-editor/helix/assets/66073333/3c8ff0c0-8313-4230-b5e2-2fbf1b9917a6">

- Is there a way around this issue?
- Can I assume that other languages that use multiple file extensions face similar issues?

**Comment by gruelingpine185** [comment]
May I ask why my issue was closed? Nobody gave context or any information regarding any of my questions nor spoke about the underlying issue so I see no reason why it's closed.

**Reply by the-mikedavis** [reply]
The GitHub UI makes the original issue look closed but it's just converted to a discussion.

We convert issues to discussions that are too broad ([Ideas](https://github.com/helix-editor/helix/discussions/categories/ideas)), bug-style issues that are not clear actionable bugs in Helix ([Troubleshooting](https://github.com/helix-editor/helix/discussions/categories/troubleshooting)) or issues lacking details (e.g. not using the bug/enhancement template, asking questions, etc.). It's a better format for exploring ideas and gathering information.

**Comment by the-mikedavis** [comment]
We try to determine the indent style from the file contents (see https://github.com/helix-editor/helix/pull/245). As @pascalkuthe mentioned (https://github.com/helix-editor/helix/issues/9082#issuecomment-1855774117) this can behave poorly on small files. I've created an enhancement-style issue to improve that here: https://github.com/helix-editor/helix/issues/9556
